docs/metrics/METRICS.md

```diff
@@ -0,0 +1,154 @@
+|------------------------------------|-------------|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| `isb_jetstream_isFull_total`       | Counter     | `buffer`=&lt;buffer-name&gt; | Indicates if the ISB is full. Continual increase of this counter metric indicates a potential backpressure that can be built on the pipeline |
+| `isb_jetstream_buffer_soft_usage`  | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the usage/utilization of a NATS Jetstream ISB                                                                                      |
+| `isb_jetstream_buffer_solid_usage` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the solid usage of a NATS Jetstream ISB                                                                                            |
+| `isb_jetstream_buffer_pending`     | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicate the number of pending messages at a given point in time.                                                                            |
+| `isb_jetstream_buffer_ack_pending` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the number of messages pending acknowledge at a given point in time                                                                |
+
+
+#### Redis ISB
+
+| Metric name              | Metric type | Labels                       | Description                                                                                                                                  |
+|--------------------------|-------------|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| `isb_redis_isFull_total` | Counter     | `buffer`=&lt;buffer-name&gt; | Indicates if the ISB is full. Continual increase of this counter metric indicates a potential backpressure that can be built on the pipeline |
+| `isb_redis_buffer_usage` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the usage/utilization of a Redis ISB                                                                                               |
+| `isb_redis_consumer_lag` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the the consumer lag of a Redis ISB                                                                                                |
+
+
+## Prometheus Operator for Scraping Metrics:
+
+You can follow the [prometheus operator](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md) setup guide if you would like to use prometheus operator configured in your cluster.
+
+### Configure the below Service Monitors for scraping your pipeline metrics:
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    name: numaflow-pipeline-metrics
```

Comment:
Use something like ` app.kubernetes.io/part-of: numaflow`?

Reply:
you mean this should be along with the name label right?